dotnet/src/Microsoft.Agents.AI.Hosting.OpenAI/ChatCompletions/Models/CreateChatCompletion.cs

```diff
@@ -0,0 +1,259 @@
+    [JsonPropertyName("prediction")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public object? Prediction { get; set; }
+
+    /// <summary>
+    /// Number between -2.0 and 2.0. Positive values penalize new tokens based on whether they appear in the text so far.
+    /// </summary>
+    [JsonPropertyName("presence_penalty")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public double? PresencePenalty { get; set; }
+
+    /// <summary>
+    /// Used by OpenAI to cache responses for similar requests to optimize your cache hit rates.
+    /// </summary>
+    [JsonPropertyName("prompt_cache_key")]
+    public string? PromptCacheKey { get; init; }
+
+    /// <summary>
+    /// The reasoning effort level for o-series models. Can be "low", "medium", or "high".
+    /// </summary>
+    [JsonPropertyName("reasoning_effort")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public string? ReasoningEffort { get; set; }
+
+    /// <summary>
+    /// An object specifying the format that the model must output.
+    /// </summary>
+    [JsonPropertyName("response_format")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public object? ResponseFormat { get; set; }
+
+    /// <summary>
+    /// A stable identifier used to help detect users of your application that may be violating OpenAI's usage policies.
+    /// The IDs should be a string that uniquely identifies each user. We recommend hashing their username or email address,
+    /// in order to avoid sending us any identifying information.
+    /// </summary>
+    [JsonPropertyName("safety_identifier")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public string? SafetyIdentifier { get; set; }
+
+    /// <summary>
+    /// If specified, the system will make a best effort to sample deterministically.
+    /// </summary>
+    [JsonPropertyName("seed")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [Obsolete("This feature is in Beta. Determinism is not guaranteed.")]
+    public long? Seed { get; set; }
+
+    /// <summary>
+    /// Specifies the processing type used for serving the request.
+    /// If set to 'auto', the request will be processed with the service tier configured in the Project settings.
+    /// If set to 'default', the request will be processed with standard pricing and performance.
+    /// If set to 'flex' or 'priority', the request will be processed with the corresponding service tier.
+    /// Defaults to 'auto'.
+    /// </summary>
+    [JsonPropertyName("service_tier")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public string? ServiceTier { get; set; }
+
+    /// <summary>
+    /// Up to 4 sequences where the API will stop generating further tokens.
+    /// </summary>
+    [JsonPropertyName("stop")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public object? Stop { get; set; }
```

Comment:
Is this the right type?

Reply:
Thought we can skip it for now, because I am not sure how we can support it in the AIAgent. Anyway, defined properly per docs